dotnet/src/Connectors/Connectors.Ollama/Services/OllamaChatCompletionService.cs

```diff
@@ -119,48 +381,403 @@ public async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessa
+                        activity.SetError(ex);
+                        throw;
+                    }
+
+                    ChatResponseStream? updateChunk = responseEnumerator.Current;
+                    if (updateChunk is null)
+                    {
+                        continue;
+                    }
+
+                    streamedRole ??= updateChunk.Message.Role;
+                    if (updateChunk is ChatDoneResponseStream doneUpdate)
+                    {
+                        streamedDoneReason ??= doneUpdate.DoneReason;
+                        metadata = GetChatCompletionMetadata(doneUpdate);
+                    }
+
+                    // If we're intending to invoke function calls, we need to consume that function call information.
+                    if (toolCallingConfig.AutoInvoke || (updateChunk.Message.ToolCalls?.Any() ?? false))
+                    {
+                        throw new NotSupportedException(
+                            "Currently, Ollama does not support function calls in streaming mode. " +
+                            "See Ollama docs at https://github.com/ollama/ollama/blob/main/docs/api.md#parameters-1 to see whether support has since been added.");
+                    }
+
+                    var streamingChatMessageContent = new StreamingChatMessageContent(GetAuthorRole(streamedRole), updateChunk.Message.Content)
+                    {
+                        Metadata = metadata,
+                        InnerContent = updateChunk,
+                        ModelId = updateChunk.Model
+                    };
+
+                    streamedContents?.Add(streamingChatMessageContent);
+                    yield return streamingChatMessageContent;
+                }
+
+                // Translate all entries into ChatCompletionsFunctionToolCall instances.
+                // toolCalls = OpenAIFunctionToolCall.ConvertToolCallUpdatesToFunctionToolCalls(
+                //     ref toolCallIdsByIndex, ref functionNamesByIndex, ref functionArgumentBuildersByIndex);
+
+                // Translate all entries into FunctionCallContent instances for diagnostics purposes.
+                // functionCallContents = this.GetFunctionCallContents(toolCalls).ToArray();
+            }
+            finally
+            {
+                activity?.EndStreaming(streamedContents, null);
+                await responseEnumerator.ConfigureAwait(false).DisposeAsync();
+            }
+        }
+    }
+
+    private ToolCallingConfig GetFunctionCallingConfiguration(Kernel? kernel, OllamaPromptExecutionSettings executionSettings, ChatHistory chatHistory, int requestIndex)
+    {
+        // If neither behavior is specified, we just return default configuration with no tool and no choice
+        if (executionSettings.FunctionChoiceBehavior is null)
+        {
+            return new ToolCallingConfig(Tools: null, Choice: null, AutoInvoke: false, AllowAnyRequestedKernelFunction: false, Options: null);
```

Comment:
The `AllowAnyRequestedKernelFunction` property is not used anywhere and can be dropped. It needs to be removed from {Azure}OpenAI connector because it's not relevant anymore but it's outside of the scope of this PR.

Reply:
Good observation, removed from the logic.